src/CMakeLists.txt

```diff
@@ -70,9 +70,18 @@ endfunction()
 
 function(valhalla_module)
   set(oneValueArgs NAME)
-  set(multiValueArgs SOURCES HEADERS DEPENDS INCLUDE_DIRECTORIES)
+  set(multiValueArgs SOURCES SOURCES_WITH_WARNINGS HEADERS DEPENDS INCLUDE_DIRECTORIES)
   cmake_parse_arguments(MODULE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
+  if (UNIX AND ENABLE_SINGLE_FILES_WERROR)
+    set_source_files_properties(
+            ${MODULE_SOURCES}
+            PROPERTIES
+            COMPILE_FLAGS
+            "-Werror -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers")
+  endif()
+  list(APPEND MODULE_SOURCES ${MODULE_SOURCES_WITH_WARNINGS})
```

Comment:
nit: I guess appending to `MODULE_SOURCES` can go into the `if`?

Reply:
I do not think so, as we need to ensure all files are compiled, regardless of the Werror configuration

Reply:
Ah right, sorry, wasn't looking into the code that followed this :sweat_smile: Then all good!